cmake/CompilerFlags.cmake

```diff
@@ -0,0 +1,453 @@
+# File which figures out the compiler flags to use based on the vendor and version of each compiler
+#Note: must be included after OpenMPConfig, MPIConfig, and PythonConfig
+
+#-------------------------------------------------------------------------------
+#  Handle CMake fortran compiler version issue
+#  See https://cmake.org/Bug/view.php?id=15372
+#-------------------------------------------------------------------------------
+	
+if(CMAKE_Fortran_COMPILER_LOADED AND "${CMAKE_Fortran_COMPILER_VERSION}" STREQUAL "")
+
+	set(CMAKE_Fortran_COMPILER_VERSION ${CMAKE_C_COMPILER_VERSION} CACHE STRING "Fortran compiler version.  May not be autodetected correctly on older CMake versions, fix this if it's wrong." FORCE)
+	message(FATAL_ERROR "Your CMake is too old to properly detect the Fortran compiler version.  It is assumed to be the same as your C compiler version, ${CMAKE_C_COMPILER_VERSION}. If this is not correct, pass -DCMAKE_Fortran_COMPILER_VERSION=<correct version> to cmake.  If it is correct,just run the configuration again.")
+	
+endif()
+	
+
+# create linker flags
+# On Windows undefined symbols in shared libraries produce errors.
+# This makes them do that on Linux too.
+set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} ${NO_UNDEFINED_FLAG}")
+
+
+
+#-------------------------------------------------------------------------------
+#  Set default flags
+#-------------------------------------------------------------------------------
+
+set(NO_OPT_FFLAGS -O0)
+set(NO_OPT_CFLAGS -O0)
+set(NO_OPT_CXXFLAGS -O0)
+
+set(OPT_FFLAGS -O3)
+set(OPT_CFLAGS -O3)
+set(OPT_CXXFLAGS -O3)
+
+set(CMAKE_C_FLAGS_DEBUG "-g")
+set(CMAKE_CXX_FLAGS_DEBUG "-g")
+set(CMAKE_Fortran_FLAGS_DEBUG "-g")
+
+#blank cmake's default optimization flags, we can't use these because not everything should be built optimized.
+set(CMAKE_C_FLAGS_RELEASE "")
+set(CMAKE_CXX_FLAGS_RELEASE "")
+set(CMAKE_Fortran_FLAGS_RELEASE "")
+
+#a macro to make things a little cleaner
+#NOTE: we can't use add_compile_options because that will apply to all languages
+macro(add_flags LANGUAGE) # FLAGS...
+	foreach(FLAG ${ARGN})
+		set(CMAKE_${LANGUAGE}_FLAGS "${CMAKE_${LANGUAGE}_FLAGS} ${FLAG}")
+	endforeach()
+endmacro(add_flags)
+
+
+#-------------------------------------------------------------------------------
+#  Now, the If Statements of Doom...
+#-------------------------------------------------------------------------------
+
+#gcc
+#---------------------------------------------------------------------------------------------------------------------------------------------------------------------
+
+if("${CMAKE_C_COMPILER_ID}" STREQUAL "GNU")
+	add_flags(C -Wall -Wno-unused-function -Wno-unknown-pragmas)
+	
+	if(NOT UNUSED_WARNINGS)
+		add_flags(C -Wno-unused-variable -Wno-unused-but-set-variable)
+	endif()
+	
+	if(NOT UNINITIALIZED_WARNINGS)
+		add_flags(C -Wno-uninitialized -Wno-maybe-uninitialized)
+	endif()
+	
+	if(${CMAKE_C_COMPILER_VERSION} VERSION_GREATER 4.1)
+		if(SSE)
```

Comment:
What's `SSE` here?

Reply:
That's an option in CompilationOptions.  For certain compilers (Intel?) it's used to select SSE optimization flags.